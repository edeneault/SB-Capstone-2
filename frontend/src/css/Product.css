.card {
  min-height: 33rem;
}

.price {
  color: rgb(140, 61, 61);
}

.list-group-item {
  min-height: 5rem;
}

.product-card:hover {
  transform: scale(1.02);
}
