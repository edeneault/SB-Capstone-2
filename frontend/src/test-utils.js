import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
