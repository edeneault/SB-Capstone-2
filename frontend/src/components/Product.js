import React from "react";
import { Link } from "react-router-dom";
import { Card, ListGroup, ListGroupItem } from "react-bootstrap";
import Rating from "./Rating";
import "../css/Product.css";

const Product = ({ product }) => {
  return (
    <Card className='my-3 rounded product-card'>
      <Link to={`/products/${product._id}`}>
        <Card.Img className='p-1 fade-in' src={product.image} variant='top' />
      </Link>
      <Card.Body>
        <ListGroup className='list-group-flush py-2'>
          <ListGroupItem>
            <Link to={`/products/${product._id}`}>
              <Card.Title className='d-block' as='div'>
                <strong> {product.name}</strong>
              </Card.Title>
            </Link>
          </ListGroupItem>
          <ListGroupItem>
            <Card.Text as='div'>
              <Rating
                value={product.rating}
                text={`${product.numReviews} reviews`}
              />
            </Card.Text>
          </ListGroupItem>

          <ListGroupItem className='price mx-auto'>
            <Card.Text as='h3'>${product.price}</Card.Text>
          </ListGroupItem>
        </ListGroup>
      </Card.Body>
    </Card>
  );
};

export default Product;
