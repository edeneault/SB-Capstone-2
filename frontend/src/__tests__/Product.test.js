import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { initialState, render } from "../test-utils";

import Product from "../components/Product";

const mockStore = configureMockStore([thunk]);

describe("test", () => {
  let product;
  beforeEach(() => {
    product = {
      rating: 5,
      numReviews: 1,
      price: 35.99,
      countInStock: 9,
      _id: "61306f7d7ef9b1460b0f6c6c",
      name: "Espresso Capsules Sampler Bundle",
      user: "6125d04db949259bdec4de35",
      image:
        "https://res.cloudinary.com/edeneault-cloudinary/image/upload/v1630786349/eShop/yt55uakwnkvkfrxck7oj.jpg",
      category: "espresso",
      brand: "Nesgresso",
      description:
        "4 Boxes of Espresso Compatible* Capsules (40 Aluminium Capsules Total)",
      reviews: [
        {
          _id: "6133398cd6359871a6a9ee28",
          name: "John Doe",
          rating: 5,
          comment: "Delicious but pricey.",
          user: "6125d04db949259bdec4de35",
          createdAt: "2021-09-04T09:17:00.324Z",
          updatedAt: "2021-09-04T09:17:00.324Z",
        },
      ],
      createdAt: "2021-09-02T06:30:21.695Z",
      updatedAt: "2021-09-09T17:44:57.014Z",
      __v: 1,
    };
  });

  test("renders Product component without crashing", () => {
    const store = mockStore(initialState);

    const { getByText } = render(
      <Provider store={store}>
        <Product product={product} />
      </Provider>,
    );

    expect(getByText("Espresso Capsules Sampler Bundle")).toBeInTheDocument();
    expect(getByText("1 reviews")).toBeInTheDocument();
    expect(getByText("$35.99")).toBeInTheDocument();
  });

  test("matches snapshot", function () {
    const store = mockStore(initialState);
    const { asFragment } = render(
      <Provider store={store}>
        <Product product={product} />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
