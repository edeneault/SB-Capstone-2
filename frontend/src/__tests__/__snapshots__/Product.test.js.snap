// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="my-3 rounded product-card card"
  >
    <a
      href="/products/61306f7d7ef9b1460b0f6c6c"
    >
      <img
        class="card-img-top p-1 fade-in"
        src="https://res.cloudinary.com/edeneault-cloudinary/image/upload/v1630786349/eShop/yt55uakwnkvkfrxck7oj.jpg"
      />
    </a>
    <div
      class="card-body"
    >
      <div
        class="list-group-flush py-2 list-group"
      >
        <div
          class="list-group-item"
        >
          <a
            href="/products/61306f7d7ef9b1460b0f6c6c"
          >
            <div
              class="d-block card-title"
            >
              <strong>
                 Espresso Capsules Sampler Bundle
              </strong>
            </div>
          </a>
        </div>
        <div
          class="list-group-item"
        >
          <div
            class="card-text"
          >
            <div
              class="rating"
            >
              <span>
                <i
                  class="fas fa-star"
                  style="color: rgb(140, 61, 61);"
                />
              </span>
              <span>
                <i
                  class="fas fa-star"
                  style="color: rgb(140, 61, 61);"
                />
              </span>
              <span>
                <i
                  class="fas fa-star"
                  style="color: rgb(140, 61, 61);"
                />
              </span>
              <span>
                <i
                  class="fas fa-star"
                  style="color: rgb(140, 61, 61);"
                />
              </span>
              <span>
                <i
                  class="fas fa-star"
                  style="color: rgb(140, 61, 61);"
                />
              </span>
              <span>
                 1 reviews
              </span>
            </div>
          </div>
        </div>
        <div
          class="price mx-auto list-group-item"
        >
          <h3
            class="card-text"
          >
            $35.99
          </h3>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
